src/runtime/free-access-test.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2018 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {FreeAccess} from './free-access';
+import {LocationProvider} from './location-provider';
+
+describes.realWin('FreeAccess', () => {
+  describe('shouldAllowFreeAccess', () => {
+    let originalReferrerDescriptor;
+
+    beforeEach(() => {
```

Comment:
nit: do we need to save the original? can we just set it to an empty string before/after?